source/MaterialXView/Viewer.cpp

```diff
@@ -1108,6 +1113,21 @@ bool Viewer::keyboardEvent(int key, int scancode, int action, int modifiers)
         return true;
     }
 
+    // Bake textures and create MTLX file for material
+    if (key == GLFW_KEY_B && action == GLFW_PRESS)
+    {
+        MaterialPtr material = getSelectedMaterial();
+        mx::DocumentPtr doc = material ? material->getDocument() : nullptr;
+        mx::TypedElementPtr elem = material ? material->getElement() : nullptr;
+        const std::string udim = material ? material->getUdim() : "";
+        _textureBaker.bakeAllInputTextures(_textureDimensions, _textureFormat, _searchPath, elem, _genContext, udim);
```

Comment:
Here I think it's important to give the user a chance to select the location to which their MaterialX document and textures will be written, so that we're not creating additional files inside their material library.  I would recommend calling the same NanoGUI method used in createSaveMaterialsInterface, giving the user a dialog that lets them choose the location of the saved MaterialX file, and then saving the baked textures to the same destination folder.